packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -862,10 +868,37 @@ export class ContainerRuntime
 			}
 		}
 
-		const idCompressorEnabled =
-			metadata?.idCompressorEnabled ?? runtimeOptions.enableRuntimeIdCompressor ?? false;
-		let idCompressor: (IIdCompressor & IIdCompressorCore) | undefined;
-		if (idCompressorEnabled) {
+		// Enabling the IdCompressor is a one-way operation and we only want to
+		// allow new containers to turn it on
+		let idCompressorMode: IdCompressorMode;
+		if (existing) {
+			// This setting has to be sticky for correctness:
+			// 1) if compressior is OFF, it can't be enabled, as already running clients (in given document session) do not know
+			//    how to process compressor ops
+			// 2) if it's ON, then all sessions should load compressor right away
+			// 3) Same logic applies for "delayed" mode
+			// Maybe in the future we will need to enabled (and figure how to do it safely) "delayed" -> "on" change.
+			// We could do "off" -> "on" transtition too, if all clients start loading compressor (but not using it initially) and do so for a while -
+			// this will allow clients to eventually to disregard "off" setting (when it's safe so) and start using compressor in future sessions.
+			// Everyting is possible, but it needs to be designed and executed carefully, when such need arises.
+			idCompressorMode = metadata?.idCompressorEnabled ?? "off";
```

Comment:
For old documents (before this change), the value of `idCompressorEnabled` is a boolean right. So, you will have to do the conversion here for those. I don't think you can simply change the type of `idCompressorEnabled` in `IContainerRuntimeMetadata` because you have to still support the old type for existing documents.
Maybe we need `ReadContainerRuntimeMetadata` and `WriteContainerRuntimeMetadata` types?

Reply:
Also, old clients reading the new value in metadata would not work right. They are expecting a boolean and will end up getting a string.
You should test this in end-to-end tests and validate that compat works in both directions.

Reply:
The feature was always marked as experimental and not used, so I do not worry that much about old documents.
That said, I'll rename it to be idCompressorMode